### ISSUE TITLE
Incorrect logits shape for GIT model (microsoft/git-base-textvqa)

### ISSUE THREAD

**Issue by souradipp76** [huggingface/transformers#33107]
### System Info

- `transformers` version: 4.42.4
- Platform: Linux-6.1.85+-x86_64-with-glibc2.35
- Python version: 3.10.12
- Huggingface_hub version: 0.23.5
- Safetensors version: 0.4.4
- Accelerate version: 0.32.1
- Accelerate config: 	not found
- PyTorch version (GPU?): 2.3.1+cu121 (True)
- Tensorflow version (GPU?): 2.17.0 (True)
- Flax version (CPU?/GPU?/TPU?): 0.8.4 (gpu)
- Jax version: 0.4.26
- JaxLib version: 0.4.26
- Using distributed or parallel set-up in script?: no
- Using GPU in script?: no
- GPU type: Tesla T4

### Who can help?

@ArthurZucker @amyeroberts @zucchini-nlp

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

I am trying to use GIT multimodal model (microsoft/git-base-textvqa) for Visual Question Answering. The shape of logits from the output of the forward function is not (batch_size, sequence_length, config.vocab_size) as mentioned in the documentation ([GIT](https://huggingface.co/docs/transformers/main/en/model_doc/git#transformers.GitForCausalLM.forward)).

1. Run the following code snippet
```
from transformers import AutoProcessor, AutoModelForCausalLM
from huggingface_hub import hf_hub_download
from PIL import Image

processor = AutoProcessor.from_pretrained("microsoft/git-base-textvqa")
model = AutoModelForCausalLM.from_pretrained("microsoft/git-base-textvqa")

file_path = hf_hub_download(repo_id="nielsr/textvqa-sample", filename="bus.png", repo_type="dataset")
image = Image.open(file_path).convert("RGB")

pixel_values = processor(images=image, return_tensors="pt").pixel_values

question = "what does the front of the bus say at the top?"

input_ids = processor(text=question, add_special_tokens=False).input_ids
input_ids = [processor.tokenizer.cls_token_id] + input_ids
input_ids = torch.tensor(input_ids).unsqueeze(0)
print(input_ids.shape) # torch.Size([1, 13])

output = model(pixel_values=pixel_values, input_ids=input_ids)
logits = output.logits
print(logits.shape) # torch.Size([1, 914, 30522])
```

### Expected behavior

The shape of the output logits should be (batch_size, sequence_length, config.vocab_size) where shape of input_ids is  (batch_size, sequence_length).

print(input_ids.shape) # torch.Size([1, 13])
...
print(logits.shape) # torch.Size([1, 13, 30522])

**Comment by NielsRogge** [comment]
Hi,

The sequence_length which comes out of GIT are the image tokens and text tokens concatenated, along with 1 special CLS token. So in this case, the 914 comes from 13 text tokens, and 914 - 13 - 1 = 900 image tokens (which you can also see by doing (`model.config.vision_config.image_size` // `model.config.vision_config.patch_size`)**2)

**Comment by souradipp76** [comment]
Thanks @NielsRogge for the clarification.
